### PR TITLE
Migrating electrodialysis reversal model

### PR DESCRIPTION
**Pull request by luohezhiming**
## Summary/Motivation:
 
Add electrodialysis reversal model (Irwin project)

## Changes proposed in this PR:
- add electrodialysis reversal model
- add corresponding tests

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by kurbansitterley — APPROVED** [review]

File: watertap/unit_models/zero_order/electrodialysis_reversal_zo.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                'TDS must be included in the solute list for determining'
+                ' electricity intensity and power consumption of the electrodialysis '
+                'reversal unit.')
+
+        self.elec_coeff_1 = Param(initialize=0.2534,
+                                  units=pyunits.kWh/pyunits.m**3,
+                                  doc="Constant 1 in electricity intensity equation")
+        self.elec_coeff_2 = Param(initialize=5.149E-4,
+                                  units=pyunits.L/pyunits.mg*pyunits.kWh/pyunits.m**3,
+                                  doc="Constant 2 in electricity intensity equation")
```
Comment (adam-a-a) [inline watertap/unit_models/zero_order/electrodialysis_reversal_zo.py:52]: following how Andrew redid the brine concentrator unit and added crystallizer as a subset with costing, we might want to follow the same changes here by (1) add these as variables here and (2) adding them to the yaml file.

**Review by adam-a-a — CHANGES_REQUESTED** [review]
Everything looks good. My only suggested changed is to take those electricity coefficient parameters and put them in the yaml. You can look at @andrewlee94 's PR where he changed some things around with the brine concentrator and crystallizer.

**Comment by adam-a-a** [comment]
FYI: this is the PR I'm referring to: https://github.com/watertap-org/watertap/pull/415

**Comment by luohezhiming** [comment]
> Everything looks good. My only suggested changed is to take those electricity coefficient parameters and put them in the yaml. You can look at @andrewlee94 's PR where he changed some things around with the brine concentrator and crystallizer.

The reason we took those coefficients as variables for the brine concentrator case is that it has a subtype and those coefficients will be different for different subtypes. For this unit, I think it is fine to use parameters.

**Comment by adam-a-a** [comment]
> > Everything looks good. My only suggested changed is to take those electricity coefficient parameters and put them in the yaml. You can look at @andrewlee94 's PR where he changed some things around with the brine concentrator and crystallizer.
> 
> The reason we took those coefficients as variables for the brine concentrator case is that it has a subtype and those coefficients will be different for different subtypes. For this unit, I think it is fine to use parameters.

The only thing is that electrodialysis will most likely have subtypes. I can think of at least one needed for AMO work, and we might also want a conventional ED subtype (without reversal). With that said, I suppose we can just merge this, and then when/if subtypes arise, we'd make that change.

**Review by adam-a-a — APPROVED** [review]

**Comment by adam-a-a** [comment]
@luohezhiming I'll approve this now, and if we need subtypes, we'll change it later.

**Comment by andrewlee94** [comment]
I would tend to agree with @adam-a-a's initial comment here. Just because we don't have different parameters yet doesn't mean we won't in future. We should avoid hard-coding parameters wherever possible (but I see I am too late now).

**Comment by luohezhiming** [comment]
@adam-a-a @andrewlee94 I will add those two parameters as variables right now.

**Pull request merged**